doc/source/reference/c-api/ufunc.rst

```diff
@@ -253,15 +253,10 @@ Functions
 .. c:function:: int PyUFunc_GenericFunction( \
         PyUFuncObject* self, PyObject* args, PyObject* kwds, PyArrayObject** mps)
 
-    A generic ufunc call. The ufunc is passed in as *self*, the arguments
-    to the ufunc as *args* and *kwds*. The *mps* argument is an array of
-    :c:type:`PyArrayObject` pointers whose values are discarded and which
-    receive the converted input arguments as well as the ufunc outputs
-    when success is returned. The user is responsible for managing this
-    array and receives a new reference for each array in *mps*. The total
-    number of arrays in *mps* is given by *self* ->nin + *self* ->nout.
-
-    Returns 0 on success, -1 on error.
+    .. deprecated:: NumPy 1.19
+
+        Unless NumPy is made aware of an issue with this, this function
+        is scheduled for rapid removal without replacement.
```

Comment:
Perhaps also add "Use `PyObject_Call(ufunc, args, kwds)` instead, and coerce the input arrays with XXX "?